### PR TITLE
Create setup.py

### PR DESCRIPTION
**Pull request by xoviat**
This will allow installation using pip.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+import os
+
+from setuptools import setup, find_packages
+
+setup(
+    name='clcache',
+    description='MSVC Complier cache',
+    author='frerich',
+    author_email='frerich@users.noreply.github.com',
```
Comment (frerich) [inline setup.py]: Please use the same email address given in the README file, i.e. raabe@froglogic.com

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+
+import os
+
+from setuptools import setup, find_packages
+
+setup(
+    name='clcache',
+    description='MSVC Complier cache',
+    author='frerich',
```
Comment (frerich) [inline setup.py]: The [examples](https://docs.python.org/3/distutils/setupscript.html) seem to use the full name here for the author, I think the code should do the same.

**Comment by frerich** [comment]
Please merge the two commits, I think there's not a lot of benefit in having two commits.

Some CI tests for this would be needed to make sure that 'pip install' really works, this would probably mean augmenting the appveyor.yml file.

**Comment by xoviat** [comment]
> Please merge the two commits, I think there's not a lot of benefit in having two commits.

You should be able to use the 'squash' method on merging.

> 'Some CI tests for this would be needed to make sure that 'pip install' really works, this would probably mean augmenting the appveyor.yml file.'

I don't think so as this is pretty well-defined behavior.

**Comment by frerich** [comment]
@xoviat Even if 'pip install' is well-defined behaviour, it would still be worthwhile to check that the `setup.py` file has the desired effect. In particular, it also serves as documentation for how to use the file. I for one never wrote such a `setup.py` file before and wouldn't even know how to use it, so seeing it in action would be good in case I document it in the Wiki and then somebody comes around and says it 'Does not work'. :-)

**Comment by frerich** [comment]
@xoviat Thanks for reminding me of the 'squash' merge option. I heard of that but never tried it myself. It sounds as if it's exactly what I wanted. :-)

File: setup.py [hunk 3]
```diff
@@ -7,8 +7,8 @@
 setup(
     name='clcache',
     description='MSVC Complier cache',
```
Comment (frerich) [inline setup.py]: There's a typo here ("Complier") and I suppose "compiler" should be written all lower-case.

File: setup.py [hunk 4]
```diff
@@ -7,8 +7,8 @@
...
-    author='frerich',
-    author_email='frerich@users.noreply.github.com',
+    author='Frerich Raabe',
+    author_email='raabe@froglogic.com',
```
Comment (frerich) [inline setup.py]: 👍

**Comment by xoviat** [comment]
You can see the install works from https://ci.appveyor.com/project/frerich/clcache/build/1.0.1065/job/xdow19mlnqqwmg9j#L136

**Comment by frerich** [comment]
@xoviat Thanks! So

```
pip install .
```

Would be how to install clcache via pip then? Does this build an executable along the way? I'm surprised that running `clcache -help` just like that works, I thought 'pip' would just copy the script somewhere.

**Comment by xoviat** [comment]
No. That command is because the git repository is already cloned. For a typical installation:

```
pip install git+https://github.com/frerich/clcache.git
```

>  Does this build an executable along the way?
Yes, but it is not a self-contained executable. It is dependent on the local python interpreter.

**Comment by xoviat** [comment]
@frerich Are there any barriers to merging this?

**Comment by frerich** [comment]
I just experimented a bit, and it seems to work for me. Thanks!

**Pull request merged**